lib/core.mli

```diff
@@ -104,254 +104,236 @@ val clone_element : element -> 'props -> element
     let clone_element element props =
       clone_element_internal Imports.react element props]
 
-val use_effect : (unit -> (unit -> unit) option_undefined) -> unit
+val use_effect_always :
+  ?before_render:bool -> (unit -> (unit -> unit) option_undefined) -> unit
   [@@js.custom
-    val use_effect_internal :
+    val _use_effect :
       Imports.react -> (unit -> (unit -> unit) option_undefined) -> unit
       [@@js.call "useEffect"]
 
-    let use_effect callback = use_effect_internal Imports.react callback]
+    val _use_layout_effect :
+      Imports.react -> (unit -> (unit -> unit) option_undefined) -> unit
+      [@@js.call "useLayoutEffect"]
 
-val use_effect0 : (unit -> (unit -> unit) option_undefined) -> unit
+    let use_effect_always ?(before_render = false) f =
+      (if before_render then _use_layout_effect else _use_effect)
+        Imports.react f]
+
+val use_effect_once :
+  ?before_render:bool -> (unit -> (unit -> unit) option_undefined) -> unit
   [@@js.custom
-    val use_effect0_internal :
+    val _use_effect :
          Imports.react
       -> (unit -> (unit -> unit) option_undefined)
-      -> Ojs.t array
+      -> 'a array
       -> unit
       [@@js.call "useEffect"]
 
-    let use_effect0 callback = use_effect0_internal Imports.react callback [||]]
-
-val use_effect1 : (unit -> (unit -> unit) option_undefined) -> 'a array -> unit
-  [@@js.custom
-    val use_effect1_internal :
+    val _use_layout_effect :
          Imports.react
       -> (unit -> (unit -> unit) option_undefined)
       -> 'a array
       -> unit
-      [@@js.call "useEffect"]
+      [@@js.call "useLayoutEffect"]
 
-    let use_effect1 callback watchlist =
-      use_effect1_internal Imports.react callback watchlist]
+    let use_effect_once ?(before_render = false) f =
+      (if before_render then _use_layout_effect else _use_effect)
+        Imports.react f [||]]
 
-val use_effect2 : (unit -> (unit -> unit) option_undefined) -> 'a * 'b -> unit
+val use_effect_on_change :
```

Comment:
I think I just wanted it for easy migration and just in case, but it's hard to think of an actual use case, yeah. I agree it can probably just be removed.

Reply:
Fixed in https://github.com/ml-in-barcelona/jsoo-react/pull/141/commits/60214607a2f36b746d22f34579dc774d01a7fc54